src/SmartyModule/Service/SmartyRendererFactory.php

```diff
@@ -39,6 +39,14 @@ public function createService(ServiceLocatorInterface $serviceLocator)
             }
         }
 
+        if (isset($config['view_manager'])
+            && isset($config['view_manager']['smarty_set_path_stack_dirs'])
+            && $config['view_manager']['smarty_set_path_stack_dirs']
```

Comment:
These two checks with isset($var) && $var doesn't make any sense, change it to 1 check !empty($var) instead.